### ISSUE TITLE
Error when calling Fortnox::API::Customer.all

### ISSUE THREAD

**Issue by ehannes** [my-codeworks/fortnox-api#22]
To reproduce the problem:

```
bundle exec irb
```

Then run the following command in irb (replace [path_to_gem]):
```
irb(main):001:0> $: << "[path_to_gem]/lib"; require 'fortnox/api'
=> true
irb(main):002:0> Fortnox::API::Customer.all
ArgumentError: bad argument (expected URI object or URI string)
        from ~/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/httparty-0.13.7/lib/httparty/request.rb:55:in `path='
        from ~/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/httparty-0.13.7/lib/httparty/request.rb:43:in `initialize'
        from ~/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/httparty-0.13.7/lib/httparty.rb:545:in `new'
        from ~/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/httparty-0.13.7/lib/httparty.rb:545:in `perform_request'
        from ~/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/httparty-0.13.7/lib/httparty.rb:476:in `get'
        from [path_to_gem]/lib/fortnox/api/base.rb:42:in `block (3 levels) in <class:Base>'
        from [path_to_gem]/lib/fortnox/api/request_handling.rb:29:in `call'
        from [path_to_gem]/lib/fortnox/api/request_handling.rb:29:in `execute'
        from [path_to_gem]/lib/fortnox/api/base.rb:41:in `block (2 levels) in <class:Base>'
        from [path_to_gem]/lib/fortnox/api/repositories/base/loaders.rb:9:in `all'
        from [path_to_gem]/lib/fortnox/api/customer.rb:18:in `all'
        from (irb):2
        from ~/.rbenv/versions/2.2.3/bin/irb:11:in `<main>'
```

**Comment by ehannes** [comment]
Calling `Fortnox::API::Customer.find(1)` gives another error message, but they could be linked:
```
irb(main):003:0> Fortnox::API::Customer.find(1)
NoMethodError: undefined method `+' for nil:NilClass
        from [path_to_gem]/lib/fortnox/api/repositories/base/loaders.rb:35:in `find_one_by'
        from [path_to_gem]/lib/fortnox/api/repositories/base/loaders.rb:28:in `find'
        from [path_to_gem]/lib/fortnox/api/customer.rb:26:in `find'
        from (irb):3
        from ~/.rbenv/versions/2.2.3/bin/irb:11:in `<main>'
```

**Comment by ehannes** [comment]
```
  1) Fortnox::API::Customer#all 
     Failure/Error: fail ArgumentError, 'You have to provide a base url.' unless base_url
     
     ArgumentError:
       You have to provide a base url.
     # ./lib/fortnox/api/input_validation.rb:9:in `validate_base_url'
     # ./lib/fortnox/api/base.rb:27:in `initialize'
     # ./lib/fortnox/api/repositories/base.rb:18:in `initialize'
     # ./lib/fortnox/api/repositories/customer.rb:32:in `initialize'
     # ./lib/fortnox/api/customer.rb:30:in `new'
     # ./lib/fortnox/api/customer.rb:30:in `repository'
     # ./lib/fortnox/api/customer.rb:18:in `all'
     # ./spec/fortnox/api/customer_spec.rb:14:in `block (3 levels) in <top (required)>'
```

In ./lib/fortnox/api/repositories/customer.rb:32, it calls `super`:
```
def initialize
  super( CONFIGURATION )
end
```
In ./lib/fortnox/api/repositories/base.rb:18, it calls super without any arguments:
```
def initialize( options )
  super()

  @options = options
end
```
In ./lib/fortnox/api/base.rb:27, initialize takes argumetns, but none were provided:
```
def initialize( base_url: nil, client_secret: nil, access_token: nil )
  base_url = validate_base_url( base_url )
  #...
end
```

**Comment by d-Pixie** [comment]
"In ./lib/fortnox/api/base.rb:27, initialize takes argumetns, but none were provided:" - This is expected. Before, as you can tell from the Readme, the gem required you to create a base Fortnox instance that you used for your calls. The arguments to the base constructor are a leftover from then.

I'm thinking we just use ENV's full on and ignore the arguments to the base constructor, or remove them since they are ignored anyway right now.

**Comment by ehannes** [comment]
Are they really ignored? Further down in that method, it sets the base URI for `HTTParty`:

`self.class.base_uri( base_url )`

**Comment by ehannes** [comment]
Now I see. You are correct. The `validate_base_url`-method reads the value from the environment. We should remove these arguments then.

**Comment by ehannes** [comment]
See PR #23.

**Comment by d-Pixie** [comment]
^ Like that (see https://github.com/my-codeworks/fortnox-api/pull/23 for the first part of this comment).

**Comment by ehannes** [comment]
Fixes #23
